File: Daily Challanges/2026-04-April/2026-04-11.py
def rook_bishop_attack(rook, bishop):
    if rook[0] == bishop[0] or rook[1] == bishop[1]:
        return "rook"
    if abs(ord(rook[0]) - ord(bishop[0])) == abs(int(rook[1]) - int(bishop[1])):
        return "bishop"
    return "neither"

print(rook_bishop_attack("A1", "A5"))
print(rook_bishop_attack("C3", "F6"))
print(rook_bishop_attack("D4", "D7"))

File: Daily Challanges/2026-04-April/2026-04-11.js
function rookBishopAttack(rook, bishop) {
  const [xRook, yRook] = [rook[0].charCodeAt(0), Number(rook[1])];
  const [xBish, yBish] = [bishop[0].charCodeAt(0), Number(bishop[1])];

  if (xRook == xBish || yRook == yBish) return "rook";
  if (Math.abs(xRook - xBish) == Math.abs(yRook - yBish)) return "bishop";
  return "neither";
}

console.log(rookBishopAttack("A1", "A5"));
console.log(rookBishopAttack("C3", "F6"));
console.log(rookBishopAttack("B3", "C5"));
